File: Sosyolobi Backend/Sosyolobi.Api/Entities/Notification.cs
using Sosyolobi.Api.Enums;

namespace Sosyolobi.Api.Entities;

public class Notification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public NotificationType Type { get; set; }
    public string Title { get; set; } = null!;
    public string? Body { get; set; }
    public Guid? RelatedActivityId { get; set; }
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: sosyolobi-mobile/lib/features/notifications/domain/notification.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../core/domain/enum_converters.dart';
import '../../../core/domain/enums.dart';

part 'notification.freezed.dart';
part 'notification.g.dart';

/// Mirrors `Sosyolobi.Api/DTOs/Notifications/NotificationResponse.cs`.
@freezed
abstract class AppNotification with _$AppNotification {
  const factory AppNotification({
    required String id,
    @NotificationTypeConverter() required NotificationType type,
    required String title,
    String? message,
    String? relatedActivityId,
    String? relatedActivityTitle,
    required bool isRead,
    required DateTime createdAt,
  }) = _AppNotification;

  factory AppNotification.fromJson(Map<String, dynamic> json) => _$AppNotificationFromJson(json);
}
